### PR TITLE
[backport] Move skl `eval_metric` and `early_stopping rounds`

### PR DESCRIPTION
**Pull request by trivialfis**
(#6751)

A new parameter `custom_metric` is added to `train` and `cv` to distinguish the behaviour from the old `feval`.  And `feval` is deprecated.  The new `custom_metric` receives transformed prediction when the built-in objective is used.  This enables XGBoost to use cost functions from other libraries like scikit-learn directly without going through the definition of the link function.

`eval_metric` and `early_stopping_rounds` in sklearn interface are moved from `fit` to `__init__` and is now saved as part of the scikit-learn model.  The old ones in `fit` function are now deprecated. The new `eval_metric` in `__init__` has the same new behaviour as `custom_metric`.

Added more detailed documents for the behaviour of custom objective and metric.

### CODE REVIEW

**Comment by trivialfis** [comment]
I will delay this to 1.6.  The change is too big to backport.
